core/src/main/java/io/grpc/internal/MessageFramer.java

```diff
@@ -249,11 +247,12 @@ private void writeBufferChain(BufferChainOutputStream bufferChain, boolean compr
     // Note that we are always delivering a small message to the transport here which
     // may incur transport framing overhead as it may be sent separately to the contents
     // of the GRPC frame.
-    sink.deliverFrame(writeableHeader, false, false);
+    sink.deliverFrame(writeableHeader, false, false, messagesBuffered);
```

Comment:
This mis-accounts the message, since messagesBuffered was already incremented for this message but this message won't be completed now. Although that doesn't really matter, it could be confusing. Seems like it deserves a comment or pass `messagesBuffered - 1` and reset with `messagesBuffered = 1`.

Reply:
I'll go with the latter, since this is meant to aid in debugging we want to be more precise.